libcalico-go/lib/ipam/ipam.go

```diff
@@ -1393,9 +1397,12 @@ func (c ipamClient) ReleaseBlockAffinity(ctx context.Context, block *model.Alloc
 		logCtx.Info("Block is already released")
 		return nil
 	}
-	hostname := getHostAffinity(block)
+	affinityCfg := getHostAffinity(block)
+	if affinityCfg == nil {
+		return nil
```

Comment:
I would assume that this branch is an error case? Is there any valid scenario where we expect this to happen?